contracts/contracts/vault/VaultAdmin.sol

```diff
@@ -246,25 +246,100 @@ contract VaultAdmin is VaultStorage {
         uint256[] calldata _amounts
     ) external onlyGovernorOrStrategist {
         require(
-            strategies[_strategyFromAddress].isSupported,
-            "Invalid from Strategy"
+            strategies[_strategyToAddress].isSupported,
+            "Invalid to Strategy"
         );
+        require(_assets.length == _amounts.length, "Parameter length mismatch");
+        _withdrawFromStrategy(
+            _strategyToAddress,
+            _strategyFromAddress,
+            _assets,
+            _amounts
+        );
+
+        IStrategy strategyTo = IStrategy(_strategyToAddress);
+        for (uint256 i = 0; i < _assets.length; i++) {
+            require(strategyTo.supportsAsset(_assets[i]), "Asset unsupported");
+        }
+        // Tell new Strategy to deposit into protocol
+        strategyTo.depositAll();
+    }
+
+    /**
+     * @dev Deposit multiple assets from the vault into the strategy.
+     * @param _strategyToAddress Address of the Strategy to deposit assets into.
+     * @param _assets Array of asset address that will be deposited into the strategy.
+     * @param _amounts Array of amounts of each corresponding asset to deposit.
+     */
+    function depositToStrategy(
+        address _strategyToAddress,
+        address[] calldata _assets,
+        uint256[] calldata _amounts
+    ) external onlyGovernorOrStrategist {
+        _depositToStrategy(_strategyToAddress, _assets, _amounts);
+    }
+
+    function _depositToStrategy(
+        address _strategyToAddress,
+        address[] calldata _assets,
+        uint256[] calldata _amounts
+    ) internal {
         require(
             strategies[_strategyToAddress].isSupported,
             "Invalid to Strategy"
         );
         require(_assets.length == _amounts.length, "Parameter length mismatch");
 
-        IStrategy strategyFrom = IStrategy(_strategyFromAddress);
         IStrategy strategyTo = IStrategy(_strategyToAddress);
 
         for (uint256 i = 0; i < _assets.length; i++) {
             require(strategyTo.supportsAsset(_assets[i]), "Asset unsupported");
-            // Withdraw from Strategy and pass other Strategy as recipient
-            strategyFrom.withdraw(address(strategyTo), _assets[i], _amounts[i]);
+            // Send required amount of funds to the strategy
+            IERC20(_assets[i]).safeTransfer(_strategyToAddress, _amounts[i]);
+            // Deposit into the Strategy from the vault
+            strategyTo.deposit(_assets[i], _amounts[i]);
```

Comment:
1. Rather than calling deposit() on the strategy each time we move a coin over, let's just call depositAll() after we have moved the funds. This is more gas efficient, but mostly importantly it can save a lot of real dollars, since depositing only a single coin into a curve pool gets costs you exchange rate fees. If we do three single coin deposits, we pay fees on each. If we deposit three coins in the same action, the fees are smaller to zero.

Reply:
that is a great point thanks